src/Literal.php

```diff
@@ -27,7 +25,7 @@ final class Literal extends Expression
     use LocationTrait;
 
     /**
-     * @param T $value
+     * @param string | int | float | bool | null | array<array-key, mixed> $value
```

Comment:
`scalar | null | array<array-key, mixed> $value`?

Reply:
I don't think there's a need to obscure the type just to make it shorter.